game/src/ecs/systems/HealthSystem.java

```diff
@@ -0,0 +1,64 @@
+package ecs.systems;
+
+import ecs.components.AnimationComponent;
+import ecs.components.HealthComponent;
+import ecs.components.MissingComponentException;
+import ecs.damage.Damage;
+import ecs.entities.Entity;
+import java.util.List;
+import mydungeon.ECS;
+
+/**
+ * The HealthSystem offsets the damage to be done to all entities with the HealthComponent. Triggers
+ * the death of an entity when the hitpoints have fallen below 0.
+ */
+public class HealthSystem extends ECS_System {
+    @Override
+    public void update() {
+        for (Entity entity : ECS.entities) {
+            entity.getComponent(HealthComponent.name)
+                    .ifPresent(
+                            component -> {
+                                {
+                                    final AnimationComponent ac =
+                                            (AnimationComponent)
+                                                    entity.getComponent(AnimationComponent.name)
+                                                            .orElseThrow(
+                                                                    () ->
+                                                                            new MissingComponentException(
+                                                                                    AnimationComponent
+                                                                                            .name));
+
+                                    HealthComponent hpComponent = (HealthComponent) component;
+
+                                    // is the entity dead?
+                                    if (hpComponent.getCurrentHitPoints() <= 0) {
+                                        hpComponent.triggerOnDeath();
+                                        ac.setCurrentAnimation(hpComponent.getDieAnimation());
+                                        // todo: After animation is finished
```

Comment:
Was will mir das TODO im Kommentar sagen? Es scheint bereits Code für diesen Fall zu existieren (die auf den Kommentar folgende Zeile)? Ist hier einfach nur vergessen worden, den Kommentar zu entfernen? Ansonsten: Weisst Du in zwei Wochen noch, was Du Dir hier sagen wolltest (und dem Rest der Welt)?

Reply:
#246

Reply:
Ja, dann schreibs doch so hin?

Reply:
Puh ne, dann formuliere ich lieber den Kommentar nochmal aus.
Issue auf Code verweisen find ich okay, aber vom Code auf ein Issue verweisen... ne danke. Das klingt schon nach Problemen.

Reply:
Naja, ist doch eh ein "Problem" aka "TODO". Da finde ich das legitim, und dann weiss der Leser, dass hier bereits für gesorgt ist. 

Ist aber auch die einzige Stelle im Code, wo ich den Verweis auf Issues als sinnvoll empfinde.